esmvaltool/diag_scripts/portrait_plot.py

```diff
@@ -0,0 +1,508 @@
+    shift = 0
+    norm = 1
+    if "mean" in method:
+        norm = array.mean(dim=dims)
+    elif "median" in method:
+        norm = array.median(dim=dims)
+    if "centered" in method:
+        shift = norm
+    normalized = (array - shift) / norm
+    return normalized
+
+
+def set_defaults(cfg):
+    """Set default values for most important config parameters."""
+    cfg.setdefault("normalize", "centered_median")
+    cfg.setdefault("x_by", "alias")
+    cfg.setdefault("y_by", "variable_group")
+    cfg.setdefault("group_by", "project")
+    cfg.setdefault("split_by", "split")  # extra facet
+    cfg.setdefault("default_split", None)
+    cfg.setdefault("cbar_kwargs", {})
+    cfg.setdefault("axes_properties", {})
+    cfg.setdefault("nan_color", 'white')
+    cfg.setdefault("figsize", (7.5, 3.5))
+    cfg.setdefault("dpi", 300)
+    cfg.setdefault("plot_legend", True)
+    cfg.setdefault("plot_kwargs", {})
+    cfg["plot_kwargs"].setdefault("cmap", "RdYlBu_r")
+    cfg["plot_kwargs"].setdefault("vmin", -0.5)
+    cfg["plot_kwargs"].setdefault("vmax", 0.5)
+    cfg.setdefault("legend", {})
+    cfg["legend"].setdefault("x_offset", 0)
+    cfg["legend"].setdefault("y_offset", 0)
+    cfg["legend"].setdefault("size", 0.3)
+
+
+def sort_data(cfg, dataset):
+    """Sort the dataset along by custom or alphabetical order."""
+    dataset = dataset.sortby([
+        dataset[cfg["x_by"]].str.lower(), dataset[cfg["y_by"]].str.lower(),
+        dataset[cfg["group_by"]].str.lower(),
+        dataset[cfg["split_by"]].str.lower()
+    ])
+    if cfg["x_by"] in ["alias", "dataset"]:
+        # NOTE: not clean, but it works for many cases
+        mm_stats = [
+            v for v in dataset[cfg["x_by"]].values
+            if "Mean" in v or "Median" in v
+        ]
+        others = [
+            v for v in dataset[cfg["x_by"]].values
+            if "Mean" not in v and "Median" not in v
+        ]
+        new_order = mm_stats + others
+        dataset = dataset.reindex({cfg["x_by"]: new_order})
+    return dataset
+
+
+def save_to_netcdf(cfg, data):
+    """Save the final dataset to a NetCDF file."""
+    basename = "portrait"
+    fname = get_diagnostic_filename(basename, cfg, extension='nc')
+    data.to_netcdf(fname)
+    log.info("NetCDF file saved:")
+    log.info(fname)
+    with ProvenanceLogger(cfg) as prov_logger:
+        prov_logger.log(fname, get_provenance(cfg))
+
+
+def main(cfg):
+    """Run the diagnostic."""
+    set_defaults(cfg)
+    metas = list(cfg["input_data"].values())
+    remove_reference(metas)
+    add_missing_facets(cfg, metas)
+    dataset = load_data(cfg, metas)
+    dataset = sort_data(cfg, dataset)
+    if cfg["normalize"] is not None:
+        dataset["var"] = normalize(dataset["var"], cfg["normalize"],
+                                   [cfg["x_by"], cfg["group_by"]])
+    plot(cfg, dataset["var"])
+    save_to_netcdf(cfg, dataset["var"])
+    with run_diagnostic() as config:
+        main(config)
```

Comment:
I just tried to run this diagnostic and realized that there is currently no output since no code is actually run when the diagnostic is executed.

```suggestion

if __name__ == '__main__':
    with run_diagnostic() as config:
        main(config)
```